addons/im_livechat/static/src/legacy/public_livechat.js

```diff
@@ -2850,692 +2850,6 @@ return Timers;
-            });
-        });
-    },
-    /**
-     * Scroll to the bottom of the thread
-     */
-    scrollToBottom: function () {
-        this.$el.scrollTop(this.el.scrollHeight);
-    },
-    /**
-     * Scrolls the thread to a given message
-     *
-     * @param {integer} options.msgID the ID of the message to scroll to
-     * @param {integer} [options.duration]
-     * @param {boolean} [options.onlyIfNecessary]
-     */
-    scrollToMessage: function (options) {
-        var $target = this.$('.o_thread_message[data-message-id="' + options.messageID + '"]');
-        if (options.onlyIfNecessary) {
-            var delta = $target.parent().height() - $target.height();
-            var offset = delta < 0 ?
-                            0 :
-                            delta - ($target.offset().top - $target.offsetParent().offset().top);
-            offset = - Math.min(offset, 0);
-            this.$el.scrollTo("+=" + offset + "px", options.duration);
-        } else if ($target.length) {
-            this.$el.scrollTo($target);
-        }
-    },
-    /**
-     * Scroll to the specific position in pixel
-     *
-     * If no position is provided, scroll to the bottom of the thread
-     *
-     * @param {integer} [position] distance from top to position in pixels.
-     *    If not provided, scroll to the bottom.
-     */
-    scrollToPosition: function (position) {
-        if (position) {
-            this.$el.scrollTop(position);
-        } else {
-            this.scrollToBottom();
-        }
-    },
-    /**
-     * Toggle all the moderation checkboxes in the thread
-     *
-     * @param {boolean} checked if true, check the boxes,
-     *      otherwise uncheck them.
-     */
-    toggleModerationCheckboxes: function (checked) {
-        this.$('.moderation_checkbox').prop('checked', checked);
-    },
-    /**
-     * Unselect the selected message
-     */
-    unselectMessage: function () {
-        this.$('.o_thread_message').removeClass('o_thread_selected_message');
-        this._selectedMessageID = null;
-    },
-
-    //--------------------------------------------------------------------------
-    // Private
-    //--------------------------------------------------------------------------
-
-    /**
-     * @private
-     */
-    _destroyOpenSeenPopoverIDs: function () {
-        _.each(this._openedSeenPopoverIDs, function (popoverID) {
-            $('#' + popoverID).remove();
-        });
-        this._openedSeenPopoverIDs = [];
-    },
-    /**
-     * Modifies $element to add the 'read more/read less' functionality
-     * All element nodes with 'data-o-mail-quote' attribute are concerned.
-     * All text nodes after a ``#stopSpelling`` element are concerned.
-     * Those text nodes need to be wrapped in a span (toggle functionality).
-     * All consecutive elements are joined in one 'read more/read less'.
-     *
-     * @private
-     * @param {jQuery} $element
-     */
-    _insertReadMore: function ($element) {
```

Comment:
maybe ci/security due to this function being moved?

Reply:
I don't see any flag at this line.